ot/gnn/_layers.py

```diff
@@ -0,0 +1,87 @@
+"""
+Template Fused Gromov Wasserstein
+"""
+
+import torch
+import torch.nn as nn
+from ._utils import template_initialisation, FGW_pooling
+
+
+class TFGWLayer(nn.Module):
+    """
+    Template Fused Gromov-Wasserstein (TFGW) layer. This layer is a pooling layer for graph neural networks.
+        It computes the fused Gromov-Wasserstein distances between the graph and a set of templates.
+
+    Parameters
+    ----------
+    n_features : int
+        Feature dimension of the nodes.
+    n_templates : int
+         Number of graph templates.
+    n_templates_nodes : int
+        Number of nodes in each template.
+    alpha0 : float, optional
+        Trade-off parameter (0 < alpha < 1). If None alpha is trained, else it is fixed at the given value.
+    train_node_weights : bool, optional
+        If True, the templates node weights are learned.
+        Else, they are uniform.
+
+
+    References
+    ----------
+    .. [52]  Cédric Vincent-Cuaz, Rémi Flamary, Marco Corneli, Titouan Vayer, Nicolas Courty.
+            "Template based graph neural network with optimal transport distances"
+    """
+
+    def __init__(self, n_features, n_templates=2, n_template_nodes=2, alpha0=None, train_node_weights=True):
+        """
+        Template Fused Gromov-Wasserstein (TFGW) layer. This layer is a pooling layer for graph neural networks.
+            It computes the fused Gromov-Wasserstein distances between the graph and a set of templates.
+
```

Comment:
add warning that alpha=logitg() tthat is optimized (to avoid constrained optimization)